packages/nodes-base/nodes/MySql/v1/GenericFunctions.ts

```diff
@@ -34,16 +34,14 @@ export async function searchTables(
 ): Promise<INodeListSearchResult> {
 	const credentials = await this.getCredentials('mySql');
 	const connection = await createConnection(credentials);
-	const sql = `
-	SELECT table_name FROM information_schema.tables
-	WHERE table_schema = '${credentials.database}'
-		and table_name like '%${query || ''}%'
-	ORDER BY table_name
-	`;
-	const [rows] = await connection.query(sql);
+	const sql =
+		'SELECT table_name FROM INFORMATION_SCHEMA.TABLES WHERE table_schema = ? and table_name like ? ORDER BY table_name';
+
+	const values = [credentials.database, `%${query ?? ''}%`];
```

Comment:
should we rename `query` to `tableNames` or something else like that to make this less confusing?

Reply:
Updated to `tableName` as it makes sense.